### PR TITLE
Add smores

### PR DESCRIPTION
**Pull request by zcarp7220**

### CODE REVIEW

File: kubejs/server_scripts/random_recipes.js [hunk 1]
```diff
@@ -554,51 +554,4 @@ ServerEvents.recipes(event => {
...
-        Item.of('gtceu:fluid_tag_filter', 1),
-        [
-            'ZZZ',
-            'ZLZ',
-            'ZZZ'
-        ], {
-        Z: 'gtceu:zinc_foil',
-        L: 'gtceu:lead_plate'
-    }
-    )
```
Comment (LetterN) [inline kubejs/server_scripts/random_recipes.js]: regression, revert this

File: kubejs/client_scripts/JEI.js [hunk 2]
```diff
@@ -89,6 +87,9 @@ JEIEvents.hideItems(event => {
 
     // Chipped
     event.hide(['chipped:botanist_workbench', 'chipped:glassblower', 'chipped:carpenters_table', 'chipped:loom_table', 'chipped:mason_table', 'chipped:alchemy_bench', 'chipped:tinkering_table'])
+
+    //Vanilla Milk
+    event.hide('minecraft:milk')
```
Comment (LetterN) [inline kubejs/client_scripts/JEI.js]: why?

File: kubejs/server_scripts/smoreRecipies.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
+ServerEvents.recipes(event => {
+// Milk Chocolate Making
+
+event.smelting('nuclearcraft:roasted_cocoa_beans', 'minecraft:cocoa_beans')
```
Comment (LetterN) [inline kubejs/server_scripts/smoreRecipies.js]:
```suggestion
event.smelting('nuclearcraft:roasted_cocoa_beans', 'minecraft:cocoa_beans').id("kubejs:cocoa_beans")
```

File: kubejs/server_scripts/smoreRecipies.js [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+event.recipes.gtceu.macerator("Nibs")
```
Comment (LetterN) [inline kubejs/server_scripts/smoreRecipies.js]:
```suggestion
event.recipes.gtceu.macerator("kubejs:ground_cocoa_nibs")
```

File: kubejs/server_scripts/smoreRecipies.js [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+    .itemInputs('nuclearcraft:roasted_cocoa_beans')
+    .itemOutputs('nuclearcraft:ground_cocoa_nibs')
+    .EUt(20)
+    .duration(100)
+
+event.recipes.gtceu.electric_blast_furnace('gtceu:chocolate_liquor')
```
Comment (LetterN) [inline kubejs/server_scripts/smoreRecipies.js]:
```suggestion
event.recipes.gtceu.electric_blast_furnace('kubejs:chocolate_liquor')
```

File: kubejs/server_scripts/smoreRecipies.js [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+    .itemInputs('nuclearcraft:ground_cocoa_nibs')
+    .itemOutputs('gtceu:tiny_carbon_dust')
+    .outputFluids(Fluid.of('nuclearcraft:chocolate_liquor', 144))
+    .duration(100)
+    .blastFurnaceTemp(307)
+    .EUt(120)
+
+event.recipes.gtceu.extractor('Butter')
```
Comment (LetterN) [inline kubejs/server_scripts/smoreRecipies.js]:
```suggestion
event.recipes.gtceu.extractor('kubejs:cocoa_butter')
```

File: kubejs/server_scripts/smoreRecipies.js [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+    .itemInputs('nuclearcraft:ground_cocoa_nibs')
+    .itemOutputs('nuclearcraft:cocoa_solids')
+    .outputFluids(Fluid.of('nuclearcraft:cocoa_butter', 144))
+    .EUt(2)
+    .duration(32)
+
+event.recipes.gtceu.mixer('nuclearcraft:unsweetened_chocolate')
```
Comment (LetterN) [inline kubejs/server_scripts/smoreRecipies.js]:
```suggestion
event.recipes.gtceu.mixer('kubejs:unsweetened_chocolate')
```

File: kubejs/server_scripts/smoreRecipies.js [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+    .inputFluids(Fluid.of('nuclearcraft:chocolate_liquor', 144), Fluid.of('nuclearcraft:cocoa_butter', 144))
+    .outputFluids(Fluid.of('nuclearcraft:unsweetened_chocolate', 288))
+    .EUt(30)
+    .duration(100)
+
+    event.recipes.gtceu.chemical_reactor('milk_pasteurization')
```
Comment (LetterN) [inline kubejs/server_scripts/smoreRecipies.js]:
```suggestion
    event.recipes.gtceu.chemical_reactor('kubejs:milk_pasteurization')
```

File: kubejs/server_scripts/smoreRecipies.js [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+    .inputFluids(Fluid.of('gtceu:milk', 1000), Fluid.of('gtceu:steam', 500))//TODO Change to forge tags for steam
+    .itemInputs('2x gtceu:stainless_steel_plate')
+    .itemOutputs('2x gtceu:stainless_steel_plate')
+    .outputFluids(Fluid.of('nuclearcraft:pasteurized_milk', 1000))
+    .EUt(480)
+    .duration(250)
+
+event.recipes.gtceu.mixer('nuclearcraft:dark_chocolate')
```
Comment (LetterN) [inline kubejs/server_scripts/smoreRecipies.js]:
```suggestion
event.recipes.gtceu.mixer('kubejs:dark_chocolate')
```

File: kubejs/server_scripts/smoreRecipies.js [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+    .inputFluids(Fluid.of('nuclearcraft:unsweetened_chocolate', 144), Fluid.of('nuclearcraft:sugar', 72))
+    .outputFluids(Fluid.of('nuclearcraft:dark_chocolate', 144))
+    .EUt(30)
+    .duration(100)
+
+event.recipes.gtceu.mixer('nuclearcraft:milk_chocolate')
```
Comment (LetterN) [inline kubejs/server_scripts/smoreRecipies.js]:
```suggestion
event.recipes.gtceu.mixer('kubejs:milk_chocolate')
```

File: kubejs/server_scripts/smoreRecipies.js [hunk 11]
```diff
@@ -0,0 +1,135 @@
...
+    .inputFluids(Fluid.of('nuclearcraft:dark_chocolate', 144), Fluid.of('nuclearcraft:pasteurized_milk', 250))
+    .outputFluids(Fluid.of('nuclearcraft:milk_chocolate',288))
+    .EUt(30)
+    .duration(80)
+
+    event.recipes.gtceu.fluid_solidifier('solid_chocolate')
```
Comment (LetterN) [inline kubejs/server_scripts/smoreRecipies.js]:
```suggestion
    event.recipes.gtceu.fluid_solidifier('kubejs:solid_chocolate')
```

File: kubejs/server_scripts/smoreRecipies.js [hunk 12]
```diff
@@ -0,0 +1,135 @@
...
+    .inputFluids(Fluid.of('nuclearcraft:milk_chocolate', 144))
+    .itemOutputs('nuclearcraft:milk_chocolate')
+    .notConsumable('gtceu:plate_casting_mold')
+    .duration(100)
+    .EUt(16)
+
+    //Grham Cracker
+event.recipes.gtceu.compressor('ghramCracker')
```
Comment (LetterN) [inline kubejs/server_scripts/smoreRecipies.js]:
```suggestion
event.recipes.gtceu.compressor('kubejs:graham_cracker')
```

File: kubejs/server_scripts/smoreRecipies.js [hunk 13]
```diff
@@ -0,0 +1,135 @@
...
+    .itemInputs('2x gtceu:wheat_dust')
+    .itemOutputs('nuclearcraft:graham_cracker')
+    .EUt(20)
+    .duration(200)
+event.recipes.gtceu.macerator('wheatToDust')
```
Comment (LetterN) [inline kubejs/server_scripts/smoreRecipies.js]:
```suggestion
event.recipes.gtceu.macerator('kubejs:wheat_dust')
```

File: kubejs/server_scripts/smoreRecipies.js [hunk 14]
```diff
@@ -0,0 +1,135 @@
...
+    .itemInputs('minecraft:wheat')
+    .itemOutputs('gtceu:wheat_dust')
+
+    //Marshmellow
+event.recipes.gtceu.fluid_solidifier('solid_marshmallow')
```
Comment (LetterN) [inline kubejs/server_scripts/smoreRecipies.js]:
```suggestion
event.recipes.gtceu.fluid_solidifier('kubejs:solid_marshmallow')
```

File: kubejs/server_scripts/smoreRecipies.js [hunk 15]
```diff
@@ -0,0 +1,135 @@
...
+    .inputFluids(Fluid.of('nuclearcraft:marshmallow', 144))
+    .itemOutputs('nuclearcraft:marshmallow')
+    .notConsumable('gtceu:ball_casting_mold')
+    .duration(100)
+    .EUt(16)
+
+event.recipes.gtceu.mixer('liquidMarshmallow')
```
Comment (LetterN) [inline kubejs/server_scripts/smoreRecipies.js]:
```suggestion
event.recipes.gtceu.mixer('kubejs:liquid_marshmallow')
```

File: kubejs/server_scripts/smoreRecipies.js [hunk 16]
```diff
@@ -0,0 +1,135 @@
...
+    .inputFluids(Fluid.of('nuclearcraft:sugar',72),Fluid.of('nuclearcraft:hydrated_gelatin',144))
+    .outputFluids(Fluid.of('nuclearcraft:marshmallow',144))
+    .EUt(30)
+    .duration(100)
+event.recipes.gtceu.extractor('liquidSugar')
```
Comment (LetterN) [inline kubejs/server_scripts/smoreRecipies.js]:
```suggestion
event.recipes.gtceu.extractor('kubejs:liquid_sugar')
```

File: kubejs/server_scripts/smoreRecipies.js [hunk 17]
```diff
@@ -0,0 +1,135 @@
...
+    .itemInputs('minecraft:sugar')
+    .outputFluids('nuclearcraft:sugar')
+    .EUt(16)
+    .duration(150)
+event.recipes.gtceu.mixer('geletanHydrate')
```
Comment (LetterN) [inline kubejs/server_scripts/smoreRecipies.js]:
```suggestion
event.recipes.gtceu.mixer('kubejs:hydrated_gelatin')
```

File: kubejs/server_scripts/smoreRecipies.js [hunk 18]
```diff
@@ -0,0 +1,135 @@
...
+    .inputFluids(Fluid.of('minecraft:water',500),Fluid.of('nuclearcraft:gelatin',144))
+    .outputFluids(Fluid.of('nuclearcraft:hydrated_gelatin',144))
+    .EUt(16)
+    .duration(45)
+event.recipes.gtceu.extractor('geltian')
```
Comment (LetterN) [inline kubejs/server_scripts/smoreRecipies.js]:
```suggestion
event.recipes.gtceu.extractor('kubejs:gelatin')
```

File: kubejs/server_scripts/smoreRecipies.js [hunk 19]
```diff
@@ -0,0 +1,135 @@
...
+    .EUt(7)
+    .duration(100)
+
+
+//Smores Finally!!!
+event.shapeless('kubejs:smore_1',[
+    '2x nuclearcraft:graham_cracker',
+    'nuclearcraft:milk_chocolate'
+    ,'nuclearcraft:marshmallow'
+])
```
Comment (LetterN) [inline kubejs/server_scripts/smoreRecipies.js]:
```suggestion
]).id('kubejs:smore_1')
```

File: kubejs/server_scripts/smoreRecipies.js [hunk 20]
```diff
@@ -0,0 +1,135 @@
...
+event.shapeless('kubejs:smore_2',[
+    '2x kubejs:smore_1',
+    'nuclearcraft:milk_chocolate'
+    ,'nuclearcraft:marshmallow'
+])
```
Comment (LetterN) [inline kubejs/server_scripts/smoreRecipies.js]:
```suggestion
]).id('kubejs:smore_2')
```

File: kubejs/server_scripts/smoreRecipies.js [hunk 21]
```diff
@@ -0,0 +1,135 @@
...
+event.shapeless('kubejs:smore_4',[
+    '2x kubejs:smore_2',
+    'nuclearcraft:milk_chocolate'
+    ,'nuclearcraft:marshmallow'
+])
```
Comment (LetterN) [inline kubejs/server_scripts/smoreRecipies.js]:
```suggestion
]).id('kubejs:smore_4')
```

File: kubejs/server_scripts/smoreRecipies.js [hunk 22]
```diff
@@ -0,0 +1,135 @@
...
+event.shapeless('kubejs:smore_8',[
+    '2x kubejs:smore_4',
+    'nuclearcraft:milk_chocolate'
+    ,'nuclearcraft:marshmallow'
+])
```
Comment (LetterN) [inline kubejs/server_scripts/smoreRecipies.js]:
```suggestion
]).id('kubejs:smore_8')
```

File: kubejs/server_scripts/smoreRecipies.js [hunk 23]
```diff
@@ -0,0 +1,135 @@
...
+event.shapeless('kubejs:smore_16',[
+    '2x kubejs:smore_8',
+    'nuclearcraft:milk_chocolate'
+    ,'nuclearcraft:marshmallow'
+])
```
Comment (LetterN) [inline kubejs/server_scripts/smoreRecipies.js]:
```suggestion
]).id('kubejs:smore_16')
```

File: kubejs/server_scripts/smoreRecipies.js [hunk 24]
```diff
@@ -0,0 +1,135 @@
...
+event.shapeless('kubejs:smore_32',[
+    '2x kubejs:smore_16',
+    'nuclearcraft:milk_chocolate'
+    ,'nuclearcraft:marshmallow'
+])
```
Comment (LetterN) [inline kubejs/server_scripts/smoreRecipies.js]:
```suggestion
]).id('kubejs:smore_32')
```

File: kubejs/server_scripts/smoreRecipies.js [hunk 25]
```diff
@@ -0,0 +1,135 @@
...
+event.shapeless('kubejs:smore_64',[
+    '2x kubejs:smore_32',
+    'nuclearcraft:milk_chocolate'
+    ,'nuclearcraft:marshmallow'
+])
```
Comment (LetterN) [inline kubejs/server_scripts/smoreRecipies.js]:
```suggestion
]).id('kubejs:smore_64')
```

**Review by LetterN — CHANGES_REQUESTED** [review]
add ids to the recipes and revert that regression

File: kubejs/client_scripts/JEI.js (hunk 2 above)
Comment (zcarp7220) [reply kubejs/client_scripts/JEI.js]: So that there only one type of milk

Comment (LetterN) [reply kubejs/client_scripts/JEI.js]: comment out the gregtech one instead, prioritize vanilla items/fluids

Comment (zcarp7220) [reply kubejs/client_scripts/JEI.js]: got it

File: kubejs/server_scripts/smoreRecipies.js [hunk 26]
```diff
@@ -0,0 +1,140 @@
...
+
+    event.recipes.gtceu.fluid_solidifier('kubejs:solidify_chocolate')
+    .inputFluids(Fluid.of('nuclearcraft:milk_chocolate', 144))
+    .itemOutputs('nuclearcraft:milk_chocolate')
+    .notConsumable('gtceu:plate_casting_mold')
+    .duration(100)
+    .EUt(16)
+
+    //Grham Cracker
+event.recipes.gtceu.compressor('ghramCracker')
```
Comment (LetterN) [inline kubejs/server_scripts/smoreRecipies.js]:
```suggestion
event.recipes.gtceu.compressor('kubejs:graham_cracker')
```

**Review by LetterN — CHANGES_REQUESTED** [review]
only missing id is this other than that pr is ok

**Pull request merged**